tests/clitests/runtests.py

```diff
@@ -19,27 +24,106 @@
 if not os.path.isabs(ENGINE_EXEC):
     os.environ['ENGINE_EXECUTABLE'] = os.path.join(CASPERJS_ROOT, ENGINE_EXEC)
 
-class TimeoutException(Exception):
-    pass
-
-
-def timeout(timeout_time):
-    def timeout_function(f):
-        def f2(*args):
-            def timeout_handler(signum, frame):
-                raise TimeoutException()
-            old_handler = signal.signal(signal.SIGALRM, timeout_handler)
-            signal.alarm(timeout_time) # triger alarm in timeout_time seconds
+def getEngine(engine_exec):
+    rawname = os.environ.get('CASPERJS_ENGINE', engine_exec)
+    rawname = os.path.basename(rawname)
+    name = re.match('^[a-zA-Z]*', rawname)
+    if None == name:
+        print("Could not get engine name from %s\n" % (rawname))
+        sys.exit(1)
+    name = name.group(0).lower()
+    cmd_args = [engine_exec, '--version']
+    version = subprocess.check_output(cmd_args).strip()
+    parts = re.match('^[^0-9]*([0-9]+)\.([0-9]+)\.([^\s])', version)
+    if None == parts:
+        print("Could not get engine version from %s\n" % (version))
+        sys.exit(1)
+    return {
+        'NAME': name,
+        'VERSION': {
+            'MAJOR': parts.group(1),
+            'MINOR': parts.group(2),
+            'PATCH': parts.group(3)
+        }
+    }
+
+ENGINE = getEngine(ENGINE_EXEC)
+
+print("ENGINE %s" % ENGINE)
```

Comment:
I'd rather prefer this not printed out to the console by default. Maybe add a dedicated option or command if you want to keep providing that feature.